application/src/main/kotlin/UserServiceImpl.kt

```diff
@@ -0,0 +1,50 @@
+/**
+ * Implementation of the UserService interface.
+ *
+ * @property userRepository the repository used for managing User entities
+ */
+class UserServiceImpl(private val userRepository: UserRepository) : UserService {
+
+    /**
+     * Creates a new user.
+     * @param user the user to create
+     * @return the created user
+     */
+    override fun createUser(user: User): User {
+        return userRepository.save(user)
+    }
```

Comment:
If you like in single-line return you can use `=`

```suggestion
    override fun createUser(user: User): User = userRepository.save(user)
```